qa/acceptance-tests/src/test/java/io/camunda/it/client/BatchOperationSearchTest.java

```diff
@@ -0,0 +1,390 @@
+
+import static io.camunda.it.util.TestHelper.deployProcessAndWaitForIt;
+import static io.camunda.it.util.TestHelper.getScopedVariables;
+import static io.camunda.it.util.TestHelper.startScopedProcessInstance;
+import static io.camunda.it.util.TestHelper.waitForBatchOperationCompleted;
+import static io.camunda.it.util.TestHelper.waitForBatchOperationWithCorrectTotalCount;
+import static io.camunda.it.util.TestHelper.waitForProcessInstanceToBeTerminated;
+import static io.camunda.it.util.TestHelper.waitForProcessesToBeDeployed;
+import static io.camunda.it.util.TestHelper.waitForScopedActiveProcessInstances;
+import static java.util.Collections.emptyMap;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.client.CamundaClient;
+import io.camunda.client.api.command.MigrationPlan;
+import io.camunda.client.api.search.enums.BatchOperationItemState;
+import io.camunda.client.api.search.enums.BatchOperationState;
+import io.camunda.client.api.search.enums.BatchOperationType;
+import io.camunda.client.api.search.response.BatchOperation;
+import io.camunda.client.api.search.response.BatchOperationItems.BatchOperationItem;
+import io.camunda.client.api.search.response.SearchResponse;
+import io.camunda.client.impl.search.filter.ProcessInstanceFilterImpl;
+import io.camunda.qa.util.multidb.MultiDbTest;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.IntStream;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.condition.DisabledIfSystemProperty;
+
+@MultiDbTest
+@DisabledIfSystemProperty(named = "test.integration.camunda.database.type", matches = "AWS_OS")
+public class BatchOperationSearchTest {
+
+  private static CamundaClient camundaClient;
+
+  private static String testScopeId;
+
+  private static String batchOperationId1;
+  private static String batchOperationId2;
+
+  private static final List<Long> ACTIVE_PROCESS_INSTANCES_1 = new ArrayList<>();
+  private static final List<Long> ACTIVE_PROCESS_INSTANCES_2 = new ArrayList<>();
+
+  @BeforeAll
+  public static void beforeAll(final TestInfo testInfo) {
+    Objects.requireNonNull(camundaClient);
+    testScopeId =
+        testInfo.getTestMethod().map(Method::toString).orElse(UUID.randomUUID().toString());
+
+    deployProcessAndWaitForIt(camundaClient, "process/service_tasks_v1.bpmn")
+        .getProcessDefinitionKey();
+    final long sourceProcessDefinitionKey =
+        deployProcessAndWaitForIt(camundaClient, "process/migration-process_v1.bpmn")
+            .getProcessDefinitionKey();
+    final var targetProcessDefinitionKey =
+        deployProcessAndWaitForIt(camundaClient, "process/migration-process_v2.bpmn")
+            .getProcessDefinitionKey();
+
+    waitForProcessesToBeDeployed(camundaClient, 3);
```

Comment:
❓ Out of curiosity: does it make sense to do "deployAndWait" three times and then do a "waitFor3Items" again?